lib/index.js

```diff
@@ -448,17 +512,15 @@ function Stream(/*optional*/xs, /*optional*/ee, /*optional*/mappingHint) {
     }
     else if (_.isObject(xs)) {
         if (_.isFunction(xs.then)) {
-            // probably a promise
-            return _(function (push) {
-                xs.then(function (value) {
-                    push(null, value);
-                    return push(null, nil);
-                },
-                function (err) {
-                    push(err);
-                    return push(null, nil);
-                });
-            });
+            //probably a promise
+            return promiseGenerator(xs);
+        }
+        else if (_.isFunction(xs.next)) {
```

Comment:
Oh, and maybe a comment here saying not to swap the order of the iterator and iterable checks.

Generators are both, and their `Symbol.iterator` returns `this`, so swapping the checks will result in an infinite loop.